tools/metaquantome/metaquantome_sample.xml

```diff
@@ -0,0 +1,112 @@
+<tool id="metaquantome_sample" name="metaQuantome: create samples file" version="@GVERSION@">
+    <description> by specifying the experiment's groups and associated column names</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <command detect_errors="exit_code"><![CDATA[
+        #set $samp_header = 'group\tcolnames'
+        echo -e '${samp_header}' > samp_file.tab;
+        #if $samps_args.samps_src == 'build':
+            #for $s in $samps_args.samps
+                #set $sample = str($s.group_name) + '\t' + str($s.col_names)
+                echo -e '${sample}' >> samp_file.tab;
+            #end for
+        #else:
+            #for $s in $samps_args.samps
+                cat ${samps_args.file} | cut -f ${s.col_names} |
+                head -n 1 | tr '\t' ',' |
+                cat <(echo -e -n "${s.group_name}\t") - >> samp_file.tab;
+            #end for
+        #end if
+    ]]></command>
+    <inputs>
+        <conditional name="samps_args">
+            <param name="samps_src" type="select" label="Sample file creation method">
+                <option value="build">Specify samples</option>
+                <option value="history">Select samples from existing file's header</option>
+            </param>
+            <when value="build">
+                <repeat name="samps" title="Samples">
+                    <param name="group_name" type="text" label="Group name" />
+                    <param name="col_names" type="text" label="Column" help="Specify the column names of the sample file. Use commas to separate multiple columns." />
+                </repeat>
+            </when>
+            <when value="history">
+                <param name="file" type="data" format="tabular" label="File with group name headers" help="Ex: moFF tool output (headers: peptide X737NS X737WS)" />
+                <repeat name="samps" title="Samples">
+                    <param name="group_name" type="text" label="Group name" />
```

Comment:
Shouldn't the group name come from the file-header?

Reply:
The help is a little bit confusing I think.

Reply:
is it better now?

Reply:
the group name isn't included in the file header - the user has to provide it.